src/hooks/useSwitchWallet.ts

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { changeActiveWallet, deriveAddressesFromStoredMetadata } from '../persistent-storage/wallets'
+import { walletUnlocked } from '../store/activeWalletSlice'
+import { ActiveWalletState } from '../types/wallet'
+import { useAppDispatch } from './redux'
+
+const useSwitchWallet = () => {
```

Comment:
It's a bit weird to me to use a hook for this. I'm not 100% sure why. 

Maybe because the only reason it's a hook is because we use `useAppDispatch`.
However, it's possible to import `dispatch` our of a component by importing the store.. So this could technically be a util function instead of a hook.

Reply:
Haha, I also didn't like it 100% when I implemented it. So I tried a different approach: To not use a hook, and just be mor explicit with dedicated dispatched actions, based on exactly what is happening inside the app. This is the code I wrote. I pushed it to a diff branch so that it's not confused with this one: https://github.com/alephium/mobile-wallet/commit/2db898e0f89e641b9fe8d92948eb06e52484a130

Reply:
What do you think of this solution? https://github.com/alephium/mobile-wallet/commit/44ae680fe9934a1cb7474ff194e3aaf0b8051013

Reply:
FWIT, discussion continued here: https://github.com/alephium/mobile-wallet/commit/44ae680fe9934a1cb7474ff194e3aaf0b8051013#r93654507